e2e_test/batch/types/uint256.slt.part

```diff
@@ -0,0 +1,405 @@
+1
+100
+10000
+100000000
+10000000000000000
+18446744073709551615
+100000000000000000000000000000000
+340282366920938463463374607431768211455
+115792089237316195423570985008687907853269984665640564039457584007913129639935
+
+query B
+select 1::rw_uint256 is null;
+----
+f
+
+query I
+select distinct 1::rw_uint256;
+----
+1
+
+# Aggregate functions
+query I
+select max(v) from t_uint256;
+----
+115792089237316195423570985008687907853269984665640564039457584007913129639935
+
+query I
+select min(v) from t_uint256;
+----
+0
+
+query I
+select count(v) from t_uint256;
+----
+10
+
+statement ok
+insert into t_uint256 values ('1');
+
+query I
+select sum(v) from t_uint256 where v <= '100000000000000000000000000000000';
+----
+100000000000018456744073809561717
+
+query I
+select count(v) from t_uint256;
+----
+11
+
+query I
+select count(distinct v) from t_uint256;
+----
+10
+
+statement ok
+drop table t_uint256;
+
+statement ok
+create table t_uint256 (v rw_uint256)
+
+statement ok
+insert into t_uint256 values ('1'), ('1'), ('2'), ('2'), ('3'), ('3'), ('4'), ('5');
+
+query RR
+select avg(v), avg(distinct v) from t_uint256;
+----
+2.625 3
+
+query RRRR
+select stddev_pop(v), stddev_samp(v), var_pop(v), var_samp(v) from t_uint256
+----
+1.3169567191065923 1.407885953173359 1.734375 1.9821428571428572
+
+statement ok
+drop table t_uint256;
+
+# Type promotion tests
+query TT
+select v::varchar, pg_typeof(v) from (select 1::rw_uint256 + 1::int) as t(v);
+----
+2 rw_uint256
+
+query TT
+select v::varchar, pg_typeof(v) from (select 1::int + 1::rw_uint256) as t(v);
+----
+2 rw_uint256
+
+query TT
+select v::varchar, pg_typeof(v) from (select 2::rw_uint256 - 1::int) as t(v);
+----
+1 rw_uint256
+
+query TT
+select v::varchar, pg_typeof(v) from (select 2::int - 1::rw_uint256) as t(v);
+----
+1 rw_uint256
+
+query TT
+select v::varchar, pg_typeof(v) from (select 1::rw_uint256 * 1::int) as t(v);
+----
+1 rw_uint256
+
+query TT
+select v::varchar, pg_typeof(v) from (select 1::int * 1::rw_uint256) as t(v);
+----
+1 rw_uint256
+
+query TT
+select v::varchar, pg_typeof(v) from (select 1::rw_uint256 / 1::int) as t(v);
+----
+1 rw_uint256
+
+query TT
+select v::varchar, pg_typeof(v) from (select 1::int / 1::rw_uint256) as t(v);
+----
+1 rw_uint256
+
+query TT
+select v::varchar, pg_typeof(v) from (select 1::rw_uint256 % 1::int) as t(v);
+----
+0 rw_uint256
+
+query TT
+select v::varchar, pg_typeof(v) from (select 1::int % 1::rw_uint256) as t(v);
+----
+0 rw_uint256
+
+query TT
+select v::varchar, pg_typeof(v) from (select sum(1::rw_uint256)) as t(v);
+----
+1 rw_uint256
+
+query TT
+select v::varchar, pg_typeof(v) from (select avg(1::rw_uint256)) as t(v);
+----
+1 double precision
+
+query TT
+select v::varchar, pg_typeof(v) from (select 1::rw_uint256 + 1::rw_uint256) as t(v);
+----
+2 rw_uint256
+
+# Edge case: subtraction that would go negative should fail
+statement error
+select 0::rw_uint256 - 1::rw_uint256;
+
+# Edge case: overflow should fail
+statement error
```

Comment:
the detailed error message can also be tested

example: https://github.com/risingwavelabs/risingwave/blob/main/e2e_test/ddl/table/generated_columns.slt.part#L17